### PR TITLE
Podman detection

### LINKED ISSUE #695: OpenRC detect RC system type podman ?
Hello,

perhaps a stupid question regarding using openrc in a container environment.
(testing a image)

I noticed that /src/librc/librc.c#L305 currently check for container=docker and so on, but podman is not detected.

cat /proc/1/environ
container=podman...

My workaround is to create a /.dockerenv to use the keyword -container feature.

Is there a better solution, or would it be a good Idea to check for container=* .

openrc -S  returns nothing without creating a /.dockerenv file.

EDIT: (alternative detections: )
podman says its provide /run/.containerenv

### PR DESCRIPTION
**Pull request by Sylviettee**
Podman is another container manager akin to Docker, being detectable by checking for the presence of `/.run/containerenv`. Fixes #695.

### CODE REVIEW

File: src/librc/librc.c [hunk 1]
```diff
@@ -304,6 +306,8 @@ detect_container(const char *systype RC_UNUSED)
 		return RC_SYS_SYSTEMD_NSPAWN;
 	else if (exists("/.dockerenv"))
 		return RC_SYS_DOCKER;
+	else if (exists("/run/.containerenv"))
+		return RC_SYS_PODMAN;
```
Comment (floppym) [inline src/librc/librc.c:310]: Please move this above the /.dockerenv check. See systemd for reference.

https://github.com/systemd/systemd/blob/v255/src/basic/virt.c#L640
https://github.com/systemd/systemd/blob/v255/src/basic/virt.c#L764-L767

File: src/librc/librc.c [hunk 2]
```diff
@@ -302,6 +304,8 @@ detect_container(const char *systype RC_UNUSED)
 		return RC_SYS_RKT;
 	else if (file_regex("/proc/1/environ", "container=systemd-nspawn"))
 		return RC_SYS_SYSTEMD_NSPAWN;
+	else if (exists("/run/.containerenv"))
```
Comment (navi-desu) [inline src/librc/librc.c:307]: *bsds don't have /run, podman docs suggest that /var/run/.containerenv will be used on freebsd containers: https://docs.podman.io/en/latest/markdown/podman-run.1.html

should we add both here? (and maybe under an ifdef)

**Review by navi-desu — CHANGES_REQUESTED** [review]

File: src/librc/librc.c (hunk 2 above)
Comment (Sylviettee) [reply src/librc/librc.c:307]: The check exists in an `#ifdef __linux__`. I could add check for `/var/run/.containerenv` under the pre-existing `#ifdef __FreeBSD__` in the function (Podman only mentions FreeBSD).

Comment (navi-desu) [reply src/librc/librc.c:312]: oh, yes. thank you

**Comment by navi-desu** [comment]
thank you!

**Pull request merged**